### DISCUSSION TITLE
Is the project maintainted?

### DISCUSSION THREAD

**Discussion by riazanovslv** [prkumar/uplink#237] [General]
Thank you for the cool package. 

I'm curious about the future of it since there's already was a release of python 3.10 and the last officialy supported version is 3.7

So are there plans to maintain this project? If I contribute and add support of python 3.10 can I expect that my pull request will be merged as well as a soon release of merged changes?

Thank you!

**Comment by prkumar** [comment]
@riazanovslv - Thanks for asking! As you mentioned, the package doesn't support versions above 3.7, mainly because I have had very little bandwidth in the last two years. With that said, if you can add support for python 3.10, I can definitely review your PR and help merge and release your change!

(I'm hoping to have more bandwidth in 2022 to dedicate to maintaining the project 😃)

**Reply by riazanovslv** [reply]
Great! So could you please point me to what is needed to be done before you can release 0.9.5? The master branch works fine on 3.10 for my workflow as well as all uplink library's tests, and I'm not sure how to contribute towards a new release.

**Reply by prkumar** [reply]
After your PR is merged, I can handle the release process for 0.9.5.

**Comment by sthagen** [comment]
@riazanovslv Maybe a PR with updated `setup.cfg` or `pyproject.toml` or other build system files list the python version dependencies?

**Reply by riazanovslv** [reply]
I can see that there is at least one not merged branch https://github.com/prkumar/uplink/pull/229 that is expected to be in 0.9.5 (though it's mostly done and only has a minor issue with coverage). 

So maybe @prkumar wants to include something else in the next release.

**Reply by prkumar** [reply]
Good catch! To avoid blocking the release of your PR, I can defer #229 to 0.9.6.

**Comment by prkumar** [comment]
I've added python 3.9 and 3.10.1 to Travis CI config: https://github.com/prkumar/uplink/commit/418bc54bc96c51c135a96c597e1b5474e62825bb

Tests are passing.

**Comment by prkumar** [comment]
I'm prepared to release v0.9.5 in the morning PST of January 3rd.

P.S. Happy (Gregorian) New Year's Day!

**Reply by riazanovslv** [reply]
Wow, it’s super cool. Thank you and Happy New Year’s Day!

**Comment by prkumar** [comment]
Following up here regarding a minor blocker for the release. See https://github.com/prkumar/uplink/pull/242#issuecomment-1004277127

Apparently, the repo has ran through all of my Travis CI credits, so builds are paused until I refresh credits. I got in touch with Travis CI Support to see if they can give open-source projects extra credits, and they mentioned they will review whether an Open Source Subscription is appropriate. I haven't heard back yet, so I'm planning on waiting one extra day to see if Support can help resolve this. (When I contacted them, it was around 6pm in Germany, which I believe is where Travis CI is headquartered.) 

If I don't hear back by EOD tomorrow, I'll purchase some credits. Alternatively, I can deploy to PyPI manually, but I would prefer to follow the automated process. I've also created #243 to consider migrating to GitHub Actions.

Apologies for the delay!

**Reply by prkumar** [reply]
Good news! Just heard back from Travis CI Support, and they gave the repo a bunch of credits so that we can resume our builds!

**Comment by prkumar** [comment]
v0.9.5 released: https://github.com/prkumar/uplink/discussions/244
